### PR TITLE
Feat/authentication

### PR DESCRIPTION
**Pull request by SHIKHASINGH1506**
1. Login, signup added 
2. custom hook for toastify

### CODE REVIEW

File: src/service/auth-service.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import axios from 'axios';
+
+const loginUser = async (loginCreds) => {
+  const loginUrl = '/api/auth/login';
+  try {
+    const { data, status } = await axios.post(loginUrl, loginCreds);
+    if (status === 200)
+      return data;
+    if (status === 500)
```
Comment (ivinayakg) [inline src/service/auth-service.js:9]: dont do 500 and 400 case, you can just let them pass down to catch

File: src/context/auth-context.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    token: '',
+    user: {},
+    isAuth: false
+  });
+  useEffect(() => {
+    setAuth({
+      token: JSON.parse(localStorage.getItem("token")),
+      user: JSON.parse(localStorage.getItem("user")),
+      isAuth: localStorage.getItem("isAuth")
+    })
```
Comment (ivinayakg) [inline src/context/auth-context.js:17]: nice thinking

File: src/pages/auth/signup.jsx [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+      if (isSignup) {
+        showToast('Account created successfuly!', 'success');
+        const { createdUser, encodedToken } = isSignup;
+        setAuth({
+          token: encodedToken,
+          user: createdUser,
+          isAuth: true
+        });
+        localStorage.setItem("token", JSON.stringify(encodedToken));
```
Comment (ivinayakg) [inline src/pages/auth/signup.jsx:42]: dont stringify the token it will give error in the future

**Review by ivinayakg — APPROVED** [review]
great work, well written, and thought-through code. 🙌
P.S - don't add files which are not related to the PR,

**Review by Rishi05051997 — APPROVED** [review]
Folder Structure Good
Reduce PR Length if possible
Clean Code

File: src/service/auth-service.js (hunk 1 above)
Comment (SHIKHASINGH1506) [reply src/service/auth-service.js:9]: I understood this and removed in my next pr.

**Pull request merged**